addons/crm/models/crm_lead.py

```diff
@@ -764,6 +761,53 @@ def action_schedule_meeting(self):
         }
         return action
 
+    def action_send_mail(self):
+        self.ensure_one()
+        template = self.env.ref('crm.email_template_opportunity_mail')
+        ctx = {
+            'default_composition_mode': 'comment',
+            'default_use_template': bool(template),
+            'default_template_id': template.id,
+        }
+        return {
+            'name': _('Send Email'),
+            'type': 'ir.actions.act_window',
+            'view_type': 'form',
+            'view_mode': 'form',
+            'res_model': 'mail.compose.message',
+            'binding_model': "crm.lead",
+            'binding_views': "list",
+            'id': "crm.action_lead_mass_mail",
+            'target': 'new',
+            'context': ctx,
+        }
+
+    def action_send_sms(self):
```

Comment:
Can't you read actions already defined in crm_sms like crm_lead_act_window_sms_composer_multi or crm_lead_act_window_sms_composer_single (funny, names are the inverse of what they should do) and simply update some missing pieces ? That way we avoid doing a whole new action by hand.